backend/tracim_backend/lib/mail_notifier/sender.py

```diff
@@ -104,10 +154,41 @@ def send_mail(self, message: MIMEMultipart):
         else:
             self.connect()  # Actually, this connects to SMTP only if required
             logger.info(self, 'Sending email to {}'.format(message['To']))
-            self._smtp_connection.send_message(message)
+            # TODO - G.M - 2019-01-29 - optimisize this code, we should not send
+            # email if connection has failed.
+            send_action = '{:8s}'.format('SENT')
+            failed_action = '{:8s}'.format('SENDFAIL')
+            action = send_action
+            try:
+                send_message_result = self._smtp_connection.send_message(message)
+                # INFO - G.M - 2019-01-29 - send_message return if not failed,
+                # dict of refused recipients.
+
+                if send_message_result == {}:
+                    logger.debug(self, 'One mail correctly sent using SMTP.')
+                else:
+                    # INFO - G.M - 2019-01-29 - send_message_result != {}
+                    # case should not happened
+                    # as we send not mail with multiple recipient at the same
+                    # time. send_message will not raise exception
+                    # just if some recipient work and some other failed.
+                    # TODO - G.M - 2019-01-29 - better support for multirecipient email
+                    logger.debug(self, 'Mail could not be send to some recipient')
```

Comment:
in case of error, you should give data to the adminsys in order to help understanding why there is an error

Reply:
you 're right, i forgot to share send_message_result